### ISSUE TITLE
Incorporate Pending in NodePhases

### ISSUE THREAD

**Issue by jessesuen** [argoproj/argo#525]
We currently transition directly from no phase to Running after a pod has been scheduled, which simplifies our node state machine. However, the Running phase hides the fact that a pod may be unschedulable, and is still Pending. This issue is to incorporate the Pending pod phase so that it is apparent that a pod was created but not yet running. 

NOTE: "virtual" nodes like step groups, would not use the Pending phase and would continue to transition directly to Running. Only leaf nodes (i.e. pods) would make use of Pending.

**Comment by jessesuen** [comment]
Example outputs of new Pending phase:
```
$ argo get pod-limits-7pxjd
Name:                pod-limits-7pxjd
Namespace:           default
ServiceAccount:      default
Status:              Running
Created:             Tue Aug 21 17:13:31 -0700 (22 seconds ago)
Started:             Tue Aug 21 17:13:31 -0700 (22 seconds ago)
Duration:            22 seconds
Parameters:
  limit:             10

STEP                    PODNAME                      DURATION  MESSAGE
 ● pod-limits-7pxjd
 ├---✔ gen-number-list  pod-limits-7pxjd-4271121940  7s
 └-·-✔ run-pod(0:1)     pod-limits-7pxjd-888908971   6s
   ├-✔ run-pod(1:2)     pod-limits-7pxjd-1445873513  7s
   ├-✔ run-pod(2:3)     pod-limits-7pxjd-885794523   6s
   ├-◷ run-pod(3:4)     pod-limits-7pxjd-1672831789  13s       ContainerCreating
   ├-◷ run-pod(4:5)     pod-limits-7pxjd-4018617467  13s       ContainerCreating
   ├-◷ run-pod(5:6)     pod-limits-7pxjd-2713948161  13s       ContainerCreating
   ├-● run-pod(6:7)     pod-limits-7pxjd-2941735403  13s
   ├-◷ run-pod(7:8)     pod-limits-7pxjd-1369060285  13s       ContainerCreating
   ├-✔ run-pod(8:9)     pod-limits-7pxjd-1321697003  9s
   └-✔ run-pod(9:10)    pod-limits-7pxjd-3434540732  8s
```

Example with ImagePullBackOff and message extracted from the pod
```
$ argo get image-pull-fail-tb57d
Name:                image-pull-fail-tb57d
Namespace:           default
ServiceAccount:      default
Status:              Running
Created:             Tue Aug 21 17:25:07 -0700 (5 seconds ago)
Started:             Tue Aug 21 17:25:07 -0700 (5 seconds ago)
Duration:            5 seconds

STEP                      PODNAME                DURATION  MESSAGE
 ◷ image-pull-fail-tb57d  image-pull-fail-tb57d  5s        ImagePullBackOff: Back-off pulling image "alpine:doesntexist"
```

**Comment by jessesuen** [comment]
Implemented in 09f0591205ec81b4ec03f0f5c534a13648346f41
